### ISSUE TITLE
Add learnable curvature for Poincaré and Lorentz models

### ISSUE THREAD

**Issue by t-yamsaki** [t-yamsaki/neembed#58]
## Goal

Make the public curvature value optionally trainable for the two supported hyperbolic geometries while preserving the existing fixed-curvature behavior by default.

Target usage should stay small, for example:

```python
model = ManifoldSentenceTransformer(
    "sentence-transformers/all-MiniLM-L6-v2",
    manifold="poincare",
    embedding_dim=64,
    curvature=1.0,
    learnable_curvature=True,
)
```

The exact public flag/name may be adjusted during implementation if a simpler API is clearly better, but avoid a general geometry-configuration framework.

## Scope

- add an opt-in learnable-curvature path for both `"poincare"` and `"lorentz"`
- keep current fixed-curvature construction unchanged when the option is disabled
- delegate the curvature parameterization and manifold math to Geoopt where available
- preserve the public meaning of `curvature` as the positive magnitude of negative sectional curvature across both geometries
- ensure the trainable representation remains positive and finite throughout optimization
- include the curvature state in `state_dict` / save-load behavior naturally rather than adding a parallel checkpoint format

## Tests

Cover at least:
- fixed-curvature behavior remains backward-compatible
- learnable curvature is a trainable parameter
- gradients reach curvature for Poincaré and Lorentz
- one optimizer step changes curvature while keeping it valid
- embeddings and geodesic distances remain finite
- save/load round-trips the learned curvature state
- public Poincaré/Lorentz curvature semantics remain consistent

## Acceptance criteria

- [ ] learnable curvature is opt-in and fixed curvature remains the default
- [ ] both Poincaré and Lorentz paths are supported
- [ ] Geoopt owns manifold math and backend parameterization details
- [ ] public curvature semantics do not become backend-specific
- [ ] learned curvature survives save/load
- [ ] normal tests and focused real-stack acceptance remain green

## Out of scope

- curvature schedules
- per-layer or per-example curvature
- mixed-curvature/product manifolds
- generic manifold registries/config objects
- trainable manifold prototypes (separate issue)
- Riemannian optimizer orchestration beyond what this issue strictly needs

## Implementation principle

Add the smallest extension that makes curvature trainable. Do not introduce a general parameter-management abstraction solely for this feature.
